subsys/suit/stream/stream_sinks/src/suit_sdfw_recovery_sink.c

```diff
@@ -219,29 +115,78 @@ static suit_plat_err_t check_urot_none(const uint8_t *buf, size_t size)
 	return err;
 }
 
-static suit_plat_err_t check_recovery_activated(const uint8_t *buf, size_t size)
+static suit_plat_err_t update_already_ongoing(const uint8_t *buf, size_t size)
 {
-	uint8_t *candidate_binary_start =
-		(uint8_t *)(buf + CONFIG_SUIT_SDFW_RECOVERY_UPDATE_FIRMWARE_OFFSET);
-	uint8_t *current_sdfw_digest = (uint8_t *)(NRF_SICR->UROT.RECOVERY.SM.TBS.FW.DIGEST);
-
-	/* Compare candidate's digest with current SDFW Recovery digest */
-	digest_sink_err_t err =
-		verify_digest(candidate_binary_start,
-			      size - (size_t)CONFIG_SUIT_SDFW_RECOVERY_UPDATE_FIRMWARE_OFFSET,
-			      PSA_ALG_SHA_512, current_sdfw_digest);
-	if (err != SUIT_PLAT_SUCCESS) {
-		if (err == DIGEST_SINK_ERR_DIGEST_MISMATCH) {
-			LOG_ERR("Digest mismatch - update failure");
-			return SUIT_PLAT_ERR_AUTHENTICATION;
-		}
+	suit_plat_err_t err = SUIT_PLAT_SUCCESS;
+
+	enum sdfw_update_status update_status = sdfw_update_initial_status_get();
 
-		LOG_ERR("Failed to calculate digest: %d", err);
-		return SUIT_PLAT_ERR_CRASH;
+	/* Candidate is different than current FW but SDFW Recovery update is already ongoing. */
+	switch (update_status) {
+	case SDFW_UPDATE_STATUS_NONE: {
+		/* No pending operation even though operation indicates SDFW Recovery update.
+		 * Yet candidate differs from current FW, so schedule the update.
+		 * TODO: Check if this situation is possible or under what circumstances
+		 */
+		err = schedule_update_and_reboot(buf, size);
+		break;
+	}
+	default: {
+		/* SecROM indicates error during update */
+		LOG_ERR("Update failure: %08x", update_status);
+		/* By design SDFW Recovery update error should not result in failing whole
+		 * installation.
+		 * Because of that, set specific error code instead of SUIT_PLAT_ERR_CRASH.
+		 */
+		err = SUIT_PLAT_ERR_SDRFW_FAILURE;
+		break;
+	}
 	}
 
-	LOG_DBG("Digest match - update success");
-	return SUIT_PLAT_SUCCESS;
+	return err;
+}
+
+static suit_plat_err_t update_needed(const uint8_t *buf, size_t size)
```

Comment:
I am not a fan of this name, it could be confused with `is_update_needed`... Perhaps `update_needed_action`?

Reply:
Changed  to update_needed_action